### PR TITLE
[WINDOWS] unskip tests

### PR DESCRIPTION
**Pull request by mattip**
## Why are these changes needed?

Unskip test files for windows, and add xfail/skip to specifically failing tests

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by mattip** [comment]
The [log has](https://github.com/ray-project/ray/runs/4175264583?check_suite_focus=true#step:8:2798) `//python/ray/tests:test_basic_2  FAILED in 3 out of 3 in 100.7s`. This passed locally. How can I tell which test is failing?

**Comment by mattip** [comment]
The test_basic_2 and test_basic_3 tests [are passing](https://github.com/ray-project/ray/runs/4182048958?check_suite_focus=true#step:8:932). @pcmoritz ping

**Review by pcmoritz — APPROVED** [review]
Thanks, this is great! There is a small linting error, can you fix it @mattip ?

```


Thu 11 Nov 2021 11:41:24 AM PST Flake8....
--
  | python/ray/tests/test_advanced.py:404:23: Q000 Remove bad quotes
  | 🚨 Error: The command exited with status 1

```

**Comment by pcmoritz** [comment]
(probably removing the print altogether is best)

**Comment by mattip** [comment]
> (probably removing the print altogether is best)

Good catch, that was not supposed to be part of this PR. Removed that change.

File: python/ray/tests/test_basic_2.py [hunk 1]
```diff
@@ -175,6 +175,7 @@ def j():
         assert ray.get(ray.get(h.remote(i))) == i
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Fails on windows")
```
Comment (czgdp1807) [inline python/ray/tests/test_basic_2.py:178]: This works on my Windows system several times. Like I ran it 20 times with pytest-repeat and it passed all the times. Any CI logs where this fails? The most probable reason I can imagine is the memory limits (1000 * 1024 * 1024) might be to large for CI.

File: python/ray/tests/test_basic_2.py [hunk 2]
```diff
@@ -240,6 +241,7 @@ def check(source_actor, dest_actor, is_large, out_of_band):
                     check(source_actor, dest_actor, is_large, out_of_band)
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Fails on windows")
```
Comment (czgdp1807) [inline python/ray/tests/test_basic_2.py:244]: This too passes with pytest-repeat 20 times.

File: python/ray/tests/test_basic_2.py (hunk 1 above)
Comment (mattip) [reply python/ray/tests/test_basic_2.py:178]: This has been merged so you can submit a new PR to enable it and see what happens.

Comment (czgdp1807) [reply python/ray/tests/test_basic_2.py:178]: I am noting things down here. Will open a PR for tests which can be unskipped without change.

File: python/ray/tests/test_basic_2.py [hunk 3]
```diff
@@ -263,6 +265,7 @@ def inc(self, count, dependency):
                     for i in range(100)]) == list(range(100))
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Fails on windows")
```
Comment (czgdp1807) [inline python/ray/tests/test_basic_2.py:268]: Passes with pytest-repeat.

File: python/ray/tests/test_basic_2.py [hunk 4]
```diff
@@ -300,6 +303,7 @@ def runner(f):
     assert delta < 10, "did not skip slow value"
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Fails on windows")
```
Comment (czgdp1807) [inline python/ray/tests/test_basic_2.py:306]: Passes with pytest-repeat.

File: python/ray/tests/test_basic_2.py [hunk 5]
```diff
@@ -620,6 +634,7 @@ def f(delay):
         ray.wait([1])
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Hangs on windows")
```
Comment (czgdp1807) [inline python/ray/tests/test_basic_2.py:637]: Passes with pytest-repeat. A bit weird. Must be a resource problem on CI.

File: python/ray/tests/test_basic_2.py [hunk 6]
```diff
@@ -699,6 +715,7 @@ def get(self):
         assert b"OK" in output
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Segfaults on windows")
```
Comment (czgdp1807) [inline python/ray/tests/test_basic_2.py:718]: Passes on pytest-repeat.

Comment (czgdp1807) [reply python/ray/tests/test_basic_2.py:718]: In non client mode though.

File: python/ray/tests/test_basic_3.py [hunk 7]
```diff
@@ -169,6 +171,7 @@ def f():
         wait_for_pid_to_exit(pid)
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Hangs on windows")
```
Comment (czgdp1807) [inline python/ray/tests/test_basic_3.py:174]: It has a performance issue. I recently bumped its time limit to 510s. Once @wuisawesome completes his PR on scheduling, then I will pick its investigation further. Please let me know if that's okay. @wuisawesome

File: python/ray/tests/test_basic_2.py [hunk 8]
```diff
@@ -354,6 +358,7 @@ def test_get_multiple(ray_start_regular_shared):
     assert results == indices
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Fails on windows")
 def test_get_with_timeout(ray_start_regular_shared):
```
Comment (czgdp1807) [inline python/ray/tests/test_basic_2.py:362]: This doesn't fail with pytest-repeat.

File: python/ray/tests/test_basic_3.py (hunk 7 above)
Comment (wuisawesome) [reply python/ray/tests/test_basic_3.py:174]: yup, i'll reenable it (actually probably can enable it now)

Comment (czgdp1807) [reply python/ray/tests/test_basic_3.py:174]: Okay. If it works then that would be great. Otherwise I am willing to investigate the causes.

**Pull request merged**